### DISCUSSION TITLE
Actions started recently failing across operating systems

### DISCUSSION THREAD

**Discussion by PhilipDeegan** [community/community#129049] [Actions] [answered]
### Select Topic Area

Bug

### Body

Our project https://github.com/PHAREHUB/PHARE is pretty complicated and we already limit the tests which run on GHA as they can require more RAM than available.

**Comment by mickeygousset** [comment]
Can you point to some failed workflows that we might take a look at?

**Reply by PhilipDeegan** [reply]
sure

same project, but forks, different branches, looks like the same issue

https://github.com/PhilipDeegan/PHARE/actions/runs/9562477144/job/26406857259
https://github.com/nicolasaunai/PHARE/actions/runs/9564439538/job/26366637384

and on macos we get `subprocess killed`

https://github.com/nicolasaunai/PHARE/actions/runs/9564439534/job/26365163349#step:13:130

you can see here that things were fine on June 13th

https://github.com/PHAREHUB/PHARE/actions

Cheers

**Reply by PhilipDeegan** [reply]
@mickeygousset is there some way I can run actions locally? we're not using any containers so my guess is no

**Reply by astevko** [reply]
The way to run actions locally is to set up a self-hosted Runner. Not necessarily for the faint of heart but it'll let you scale beyond the basics. 

The subprocess killed message can be due to step timeout - look for a message or the execution time for a step that is out of alignment with the norm baseline.

**Reply by PhilipDeegan** [reply]
@astevko I do see a test that is timing out, and it's the same test across all operating systems.
We are not able to reproduce this locally so I figure it's some recent change in how github actions are working as all operating systems started failing at the same time

**Reply by astevko** [reply]
Timing out at the default 10m or did you override the timeout-minutes: directive?

I find that trying to benchmark actions locally is a mixed bag - the 2 slow CPUs and 7 GiB ram stock Azure VM is difficult to compare with my many-core dedicated development machine.

**Reply by PhilipDeegan** [reply, marked as answer]
@astevko @mickeygousset I think I've tracked down the issue with the help of  https://github.com/nektos/act

it seems that `ifstream::eof()` does not work anymore, and we are OOM-ing during some file read operations

**Reply by mickeygousset** [reply]
interesting....

**Reply by Xf9s** [reply]
F

**Comment by PhilipDeegan** [comment]
@astevko  @mickeygousset 

while I have you here, can you please tell me if this is something that can be implemented:
https://github.com/orgs/community/discussions/124141 https://github.com/actions/runner/issues/3337

**Comment by 07SD** [comment]
![MyLINK-trackingnumber.txt](https://github.com/user-attachments/files/16021343/MyLINK-trackingnumber.txt)
